File: backend/src/main/java/no/strazdins/rebus/dto/TeamAnswerDto.java
package no.strazdins.rebus.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import java.util.List;

/**
 * Data-transfer object for answers to all challenges for a particular team.
 */
@Schema(description = "Answers to all challenges for a particular team")
public record TeamAnswerDto(
    @Schema(description = "ID of the team", example = "1")
    int teamId,
    @Schema(description = "List of answers to all challenges for the team")
    List<AnswerDto> answers
) {
}

File: backend-v2/src/common/types/dto/teamAnswerDto.ts
import { AnswerDto } from "./answerDto";
import { HttpResponseDto } from "./httpResponse";

/**
 * Team answer DTO.
 *
 * @property teamId ID of the team
 * @property answers Answers of the team
 */
export type TeamAnswerDto = {
  teamId: number;
  answers: AnswerDto[];
};

/**
 * Answers for one team.
 */
export type TeamAnswerResponseDto = HttpResponseDto<TeamAnswerDto>;
/**
 * Answers for all teams.
 */
export type TeamListAnswerResponseDto = HttpResponseDto<TeamAnswerDto[]>;
